app/routes/datasets/dataset.py

```diff
@@ -28,13 +28,15 @@
 async def get_owner(
     dataset: str = Depends(dataset_dependency), user: User = Depends(get_manager)
 ) -> User:
-    """Retrieves the user object that owns the dataset if that user is the one
-    making the request, otherwise raises a 401."""
+    """Returns the User making the request as long as that user is an admin or
+    the owner of the dataset, otherwise raises a 401."""
 
+    if user.role == "ADMIN":
+        return user
```

Comment:
Thanks for catching this, I thought I had it in the code but seem to have lost it. Justin caught it too.

Reply:
I'll make mine match this